packages/environment-ember-loose/__tests__/type-tests/glimmer-component.test.ts

```diff
@@ -63,17 +64,27 @@ import { ComponentLike } from '@glint/template';
   }
 
   class YieldingComponent<T> extends Component<YieldingComponentSignature<T>> {
-    static template = template(function* <T>(𝚪: ResolveContext<YieldingComponent<T>>) {
-      expectTypeOf(𝚪.this).toEqualTypeOf<YieldingComponent<T>>();
-      expectTypeOf(𝚪.args).toEqualTypeOf<{ values: T[] }>();
-      expectTypeOf(𝚪.this.args).toEqualTypeOf<Readonly<{ values: T[] }>>();
-
-      if (𝚪.args.values.length) {
-        yieldToBlock(𝚪, 'default', 𝚪.args.values[0]);
-      } else {
-        yieldToBlock(𝚪, 'else');
-      }
-    });
+    static {
+      templateForBackingValue(this, function* (𝚪) {
+        // We can't directly assert on the type of e.g. `@values` here, as we don't
+        // have a name for it in scope. However, the yields below confirm that the
+        // only thing we can legally yield to the default block is an element of the
+        // `@values` arg, since the only information we have about that type is that
+        // the array element and the yielded value are the same.
```

Comment:
I may be missing something obvious, but couldn't we minimally check something like this?

```ts
expectTypeOf(Γ.args).toEqualTypeOf<{ values: Array<T> }>();
```

I can't see how `Γ.args.values` works below otherwise! Again, maybe I’m missing something, though?

Reply:
There's no `T` we can reference, because type parameters on classes are only in scope within their instance properties and methods, not static ones. That's what I was aiming to get at with `we don't have a name for it in scope`, but I'm happy to adjust if you can suggest better phrasing.

Reply:
I *was* missing something, see comment below. DOH. (I meant to remove this.)